src/Engine/ProtoCore/DSASM/InstructionSet.cs

```diff
@@ -362,6 +342,258 @@ public bool IsReferenceType
+        {
+            get
+            {
+                Check(IsDouble, "The Type of StackValue is not Double");
+                return BitConverter.Int64BitsToDouble(opdata);
+            }
+        }
+
+        public bool BooleanValue
+        {
+            get
+            {
+                Check(IsBoolean, "The Type of StackValue is not Boolean");
+                return opdata != 0;
+            }
+        }
+
+        public int StringPointer
+        {
+            get
+            {
+                Check(IsString, "The Type of StackValue is not String");
+                return (int)opdata;
+            }
+        }
+
+        public int LabelIndex
+        {
+            get
+            {
+                Check(IsLabelIndex, "The Type of StackValue is not LabelIndex");
+                return (int)opdata;
+            }
+        }
+
+        public int BlockIndex
+        {
+            get
+            {
+                Check(IsBlockIndex, "The Type of StackValue is not BlockIndex");
+                return (int)opdata;
+            }
+        }
+
+        public int Pointer
+        {
+            get
+            {
+                Check(IsPointer, "The Type of StackValue is not Pointer");
+                return (int)opdata;
+            }
+        }
+
+        public int ArrayPointer
+        {
+            get
+            {
+                Check(IsArray, "The Type of StackValue is not ArrayPointer");
+                return (int)opdata;
+            }
+        }
+
+        public int FunctionPointer
+        {
+            get
+            {
+                Check(IsFunctionPointer, "The Type of StackValue is not FunctionPointer");
+                return (int)opdata;
+            }
+        }
+
+        public int ArrayDimension
+        {
+            get
+            {
+                Check(IsArrayDimension, "The Type of StackValue is not ArrayDimension");
+                return (int)opdata;
+            }
+        }
+
+        public int ReplicationGuide
+        {
+            get
+            {
+                Check(IsReplicationGuide, "The Type of StackValue is not ReplicationGuide");
+                return (int)opdata;
+            }
+        }
+
+        public int Rank 
+        {
+            get
+            {
+                Check(IsStaticType, "The Type of StackValue is not StaticType");
+                return (int)opdata;
+            }
+        }
+
+        public CallingConvention.CallType CallType
+        {
+            get
+            {
+                Check(IsCallingConvention, "The Type of StackValue is not CallingConvention");
+                return (CallingConvention.CallType)opdata;
+            }
+        }
+
+        public CallingConvention.BounceType BounceType 
+        {
+            get
+            {
+                Check(IsCallingConvention, "The Type of StackValue is not CallingConvention");
+                return (CallingConvention.BounceType)opdata;
```

Comment:
I'm doing refactoring: creating a new AddressType CallType and BounceType, instead of mixing them together. On the way! :-)